compiler/modules/front.syntax/astString.bal

```diff
@@ -284,7 +284,10 @@ function typeDescToWords(Word[] w, TypeDesc td, boolean|BinaryTypeOp wrap = fals
         if wrap != false {
             w.push("(");
         }
-        typeDescToWords(w, td.rest, true);
+        TypeDesc? rest = td.rest;
+        if rest != () {
+            typeDescToWords(w, rest, true);
+        }
         w.push(CLING);
         w.push("[", "]");
```

Comment:
I think you'll need to serialize length here, please add some test cases to `testParser.json`.

Reply:
ahh didn't notice that, I'll fix this and add test cases like `int[x] v = xxx` to `testParser.json`